### PR TITLE
Geomap: Update docs for 10.1

### PR DESCRIPTION
**Pull request by nmarrs**
Update geomap docs for 10.1

Remaining tasks:

- [x] Add network map layer docs
- [x] Add network map layer media (image / gif etc) for docs

### CODE REVIEW

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 1]
```diff
@@ -250,6 +274,112 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+  - **None**
+  - **Forward**
+  - **Reverse**
+  - **Both**
+- **Show legend** allows you to toggle the legend for the layer. **Note:** The legend currently only supports node data.
+- **Display tooltip** allows you to toggle tooltips for the layer.
+
+#### Node styles
+
+- **Node Size** configures the size of the node. The default is `Fixed size`, which makes all node sizes the same regardless of the data points. However, there is also an option to scale the nodes to the corresponding data points. `Min` and `Max` node size has to be set such that the nodes can scale within this range.
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:370]:
```suggestion
- **Node Size** configures the size of the node. The default is `Fixed size`, which makes all node sizes the same regardless of the data points. However, there is also an option to scale the nodes to the corresponding data points. `Min` and `Max` node sizes have to be set such that the nodes can scale within this range.
```

Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:370]: I don't see this layer in the test instance I'm in, but please confirm that these UI labels are actually "Node Size", "Node Shape", and "Node Color" vs just "Size", "Shape", and "Color".

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 2]
```diff
@@ -250,6 +274,112 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+- **Node Shape** allows you to choose the shape, icon, or graphic to aid in providing additional visual context to your data. Choose from assets that are included with Grafana such as simple shapes or the Unicon library. You can also specify a URL containing an image asset. The image must be a scalable vector graphic (SVG).
+- **Node Color** configures the color of the node. The default `Single color` keeps all points a single color. There is an alternate option to have multiple colors depending on the data point values and the threshold set at the `Thresholds` section.
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:372]:
```suggestion
- **Node Color** configures the color of the node. The default `Single color` keeps all points a single color. There is an alternate option to have multiple colors depending on the data point values and the threshold set in the `Thresholds` section.
```

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 3]
```diff
@@ -250,6 +274,112 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+- **Fill opacity** configures the transparency of each node.
+- **Rotation angle** configures the rotation angle of each node. The default is `Fixed size`, which makes all nodes rotate to the same angle the same regardless of the data points. However, there is also an option to rotate the nodes tied to field data.
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:374]:
```suggestion
- **Rotation angle** configures the rotation angle of each node. The default is `Fixed size`, which makes all nodes rotate to the same angle regardless of the data points. However, there is also an option to rotate the nodes tied to field data.
```

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 4]
```diff
@@ -250,6 +274,112 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+- **Text label** configures a text label for each node.
+
+#### Edge styles
+
+- **Edge Size** configures the size of the edge. The default is `Fixed size`, which makes all edge sizes the same regardless of the data points. However, there is also an option to scale the edges to the corresponding data points. `Min` and `Max` edge size has to be set such that the edges can scale within this range.
+- **Edge Color** configures the color of the edge. The default `Single color` keeps all points a single color. There is an alternate option to have multiple colors depending on the data point values and the threshold set at the `Thresholds` section.
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:380]: Same note as previous about UI names.

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 5]
```diff
@@ -250,6 +274,112 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+- **Node Color** configures the color of the node. The default `Single color` keeps all points a single color. There is an alternate option to have multiple colors depending on the data point values and the threshold set at the `Thresholds` section.
+- **Fill opacity** configures the transparency of each node.
+- **Rotation angle** configures the rotation angle of each node. The default is `Fixed size`, which makes all nodes rotate to the same angle the same regardless of the data points. However, there is also an option to rotate the nodes tied to field data.
+- **Text label** configures a text label for each node.
+
+#### Edge styles
+
+- **Edge Size** configures the size of the edge. The default is `Fixed size`, which makes all edge sizes the same regardless of the data points. However, there is also an option to scale the edges to the corresponding data points. `Min` and `Max` edge size has to be set such that the edges can scale within this range.
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:379]:
```suggestion
- **Edge Size** configures the size of the edge. The default is `Fixed size`, which makes all edge sizes the same regardless of the data points. However, there is also an option to scale the edges to the corresponding data points. `Min` and `Max` edge sizes have to be set such that the edges can scale within this range.
```

**Review by imatwawana — COMMENTED** [review]
This is great! I've made a number of wording tweaks and left a few questions on things that aren't clear to me.

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 6]
```diff
@@ -250,6 +274,112 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+
+## Route layer (Beta)
+
+The Route layer renders data points as a route.
+
+{{< figure src="/media/docs/grafana/geomap-route-layer-basic-9-4-0.png" max-width="1200px" caption="Geomap panel Route" >}}
+
+### Options
+
+- **Size** sets the route thickness. Fixed by default, or Min and Max range of selected field.
```
[the remark this answers is not in the document]
Comment (nmarrs) [reply docs/sources/panels-visualizations/visualizations/geomap/index.md]: hm it seems in this case that we only allow setting the min / max if user has selected field data - I'll update docs accordingly

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 7]
```diff
@@ -250,6 +274,112 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+- **Color** sets the route color. Fixed by default or Standard Options color scheme on selected field.
```
[the remark this answers is not in the document]
Comment (nmarrs) [reply docs/sources/panels-visualizations/visualizations/geomap/index.md]: eh including standard options mention here doesn't make a ton of sense given user can set fixed color or set color based on data... I think we can just remove the reference to color schemes

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 8]
```diff
@@ -250,6 +278,125 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+
+- [**Extensions for OpenLayers - Image Photo Style**](http://viglino.github.io/ol-ext/examples/style/map.style.photo.html)
+
+## Network layer (Beta)
+
+{{% admonition type="caution" %}}
+The Network layer is currently in [public preview](/docs/release-life-cycle/). Grafana Labs offers limited support, and breaking changes might occur prior to the feature being made generally available.
+{{% /admonition %}}
+
+The Network layer renders a network graph. This layer supports the same [data format supported by the node graph visualization]({{< relref "../node-graph/#data-api" >}}) with the addition of [geospatial data]({{< relref "#location">}}) included the nodes data. The geospatial data is used to render the nodes on the map.
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:370]:
```suggestion
The Network layer renders a network graph. This layer supports the same [data format supported by the node graph visualization]({{< relref "../node-graph/#data-api" >}}) with the addition of [geospatial data]({{< relref "#location">}}) included in the nodes data. The geospatial data is used to locate and render the nodes on the map.
```

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 9]
```diff
@@ -250,6 +278,125 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+
+{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-geomap-network-layer-v2.png" max-width="750px" caption="Geomap network layer" >}}
+{{< video-embed src="/media/docs/grafana/screen-recording-10-1-geomap-network-layer-from-node-graph.mp4" max-width="750px" caption="Node graph to Geomap network layer" >}}
+
+### Options
+
+- **Arrow** sets the arrow styling to display for each edge. Choose from:
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:377]:
```suggestion
- **Arrow** sets the arrow direction to display for each edge, with forward meaning source to target. Choose from:
```

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 10]
```diff
@@ -250,6 +278,125 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+  - **None**
+  - **Forward**
+  - **Reverse**
+  - **Both**
+- **Show legend** allows you to toggle the legend for the layer. **Note:** The legend currently only supports node data.
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:382]: +1

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 11]
```diff
@@ -250,6 +278,125 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+- **Display tooltip** allows you to toggle tooltips for the layer.
+
+#### Node styles
+
+- **Size** configures the size of the node. The default is `Fixed size`, which makes all node sizes the same regardless of the data points. However, there is also an option to scale the nodes to the corresponding data points. `Min` and `Max` node sizes have to be set such that the nodes can scale within this range.
+- **Symbol** allows you to choose the symbol, icon, or graphic to aid in providing additional visual context to your data. Choose from assets that are included with Grafana such as simple symbols or the Unicon library. You can also specify a URL containing an image asset. The image must be a scalable vector graphic (SVG).
+- **Color** configures the color of the node. The default `Single color` keeps all points a single color. There is an alternate option to have multiple colors depending on the data point values and the threshold set in the `Thresholds` section.
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:389]:
```suggestion
- **Color** configures the color of the nodes. The default `Fixed color` sets all nodes to a specific color. There is also an option to have conditional colors depending on the selected field data point values and the color scheme set in the `Standard options` section.
```

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 12]
```diff
@@ -250,6 +278,125 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+- **Fill opacity** configures the transparency of each node.
+- **Rotation angle** configures the rotation angle of each node. The default is `Fixed size`, which makes all nodes rotate to the same angle regardless of the data points. However, there is also an option to rotate the nodes tied to field data.
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:391]:
```suggestion
- **Rotation angle** configures the rotation angle of each node. The default is `Fixed value`, which makes all nodes rotate to the same angle regardless of the data; however, there is also an option to set the rotation of the nodes based on data corresponding to a selected field.
```

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 13]
```diff
@@ -250,6 +278,125 @@ The GeoJSON layer allows you to select and load a static GeoJSON file from the f
...
+- **Text label** configures a text label for each node.
+
+#### Edge styles
+
+- **Size** configures the size of the edge. The default is `Fixed size`, which makes all edge sizes the same regardless of the data points. However, there is also an option to scale the edges to the corresponding data points. `Min` and `Max` edge sizes have to be set such that the edges can scale within this range.
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:396]:
```suggestion
- **Size** configures the line width of the edges. The default is `Fixed size`, which makes all edge line widths the same regardless of the data; however, there is also an option to size the edges based on data corresponding to a selected field. `Min` and `Max` eges sizes have to be set such that the edges can scale within this range.
```

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 14]
```diff
@@ -215,10 +237,14 @@ The markers layer allows you to display data points as different marker shapes s
...
-- **Marker Size** configures the size of the marker. The default is `Fixed size`, which makes all marker sizes the same regardless of the data points. However, there is also an option to scale the circles to the corresponding data points. `Min` and `Max` marker size has to be set such that the Marker layer can scale within this range.
-- **Marker Shape** allows you to choose the shape, icon, or graphic to aid in providing additional visual context to your data. Choose from assets that are included with Grafana such as simple shapes or the Unicon library. You can also specify a URL containing an image asset. The image must be a scalable vector graphic (SVG).
+- **Size** configures the size of the marker. The default is `Fixed size`, which makes all marker sizes the same regardless of the data points. However, there is also an option to scale the markers to the corresponding data points. `Min` and `Max` marker sizes have to be set such that the Marker layer can scale within this range.
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:240]:
```suggestion
- **Size** configures the size of the markers. The default is `Fixed size`, which makes all marker sizes the same regardless of the data; however, there is also an option to size the markers based on data corresponding to a selected field. `Min` and `Max` marker sizes have to be set such that the markers can scale within this range.
```

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 15]
```diff
@@ -215,10 +237,14 @@ The markers layer allows you to display data points as different marker shapes s
...
+- **Symbol** allows you to choose the symbol, icon, or graphic to aid in providing additional visual context to your data. Choose from assets that are included with Grafana such as simple symbols or the Unicon library. You can also specify a URL containing an image asset. The image must be a scalable vector graphic (SVG).
+- **Color** configures the color of the marker. The default `Single color` keeps all points a single color. There is an alternate option to have multiple colors depending on the data point values and the threshold set in the `Thresholds` section.
 - **Fill opacity** configures the transparency of each marker.
+- **Rotation angle** configures the rotation angle of each marker. The default is `Fixed size`, which makes all markers rotate to the same angle regardless of the data points. However, there is also an option to rotate the markers tied to field data.
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:244]:
```suggestion
- **Rotation angle** configures the rotation angle of each marker. The default is `Fixed value`, which makes all markers rotate to the same angle regardless of the data; however, there is also an option to set the rotation of the markers based on data corresponding to a selected field.
```

**Comment by drew08t** [comment]
Just some minor re-wording suggested (applied to nodes, edges, and markers for consistency), otherwise, looks great!

**Review by drew08t — APPROVED** [review]

**Review by imatwawana — APPROVED** [review]
I'm good with the two suggestions you made. This LGTM!

**Comment by imatwawana** [comment]
Attempting to retroactively backport these docs to 10.1. The day we merged these docs, the new branch was cut, so these updates didn't make it into 10.1 docs.

**Pull request merged**